stock_valuation_fifo_lot/models/product.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2023 Ecosoft Co., Ltd (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+from odoo.tools import float_is_zero
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _sort_by_all_candidates(self, all_candidates, sort_by):
+        """Hook function for other sort by"""
+        return all_candidates
+
+    def _get_all_candidates(self, company, sort_by=None):
+        all_candidates = (
+            self.env["stock.valuation.layer"]
+            .sudo()
+            .search(
+                [
+                    ("product_id", "=", self.id),
+                    ("remaining_qty", ">", 0),
+                    ("company_id", "=", company.id),
+                ]
+            )
+        )
+        if sort_by == "lot_create_date":
+
+            def sorting_key(candidate):
+                if len(candidate.lot_ids) > 1:
+                    return min(candidate.lot_ids.mapped("create_date"))
+                elif candidate.lot_ids:
+                    return candidate.lot_ids[0].create_date
+                else:
+                    return candidate.create_date
+
+            all_candidates = all_candidates.sorted(key=sorting_key)
+        elif sort_by is not None:
+            all_candidates = self._sort_by_all_candidates(all_candidates, sort_by)
+        return all_candidates
+
+    def _run_fifo(self, quantity, company):
+        self.ensure_one()
+        move_id = self._context.get("used_in_move_id")
+        if self.tracking == "none" or not move_id:
+            vals = super()._run_fifo(quantity, company)
```

Comment:
```suggestion
            return super()._run_fifo(quantity, company)
```

and then remove the else ladder to save some space and improve the readability.